classes/profile.php

```diff
@@ -53,160 +59,119 @@ class profile {
     const SCRIPTTYPE_WS = 3;
     const SCRIPTTYPE_TASK = 4;
 
-    const DENYLIST = [
-        manager::MANUAL_PARAM_NAME,
-        manager::FLAME_ON_PARAM_NAME,
-        manager::FLAME_OFF_PARAM_NAME,
-    ];
 
-    const REDACTLIST = [
-        'sesskey',
-    ];
-
-    /**
-     * Stores the ID of a saved profile, to indicate that it should be overwritten.
-     *
-     * @var int
-     */
-    public static $partialsaveid = 0;
-
-    /**
-     * Removes any parameter on profile::DENYLIST.
-     *
-     * @param array $parameters
-     * @return array
-     */
-    public static function stripparameters(array $parameters): array {
-        $parameters = array_filter(
-            $parameters,
-            function($i) {
-                return !in_array($i, self::DENYLIST);
-            },
-            ARRAY_FILTER_USE_KEY
-        );
-
-        foreach ($parameters as $i => &$v) {
-            if (in_array($i, self::REDACTLIST)) {
-                $v = '';
-            }
-        }
+    private static $runningprofile = null;
 
-        return $parameters;
+    // TODO: try to find a way to eliminate the need for this function.
+    public static function get_num_profiles(): int {
+        global $DB;
+        return $DB->count_records(self::TABLE, []);
     }
 
     /**
-     * Gets a single profile, including data.
+     * Loads a profile from disk.
```

Comment:
from db, not disk

Reply:
The function has been erased as redundant.